man/sd_bus_service_reconnect.c

```diff
@@ -92,6 +92,24 @@ static int on_disconnect(sd_bus_message *message, void *userdata, sd_bus_error *
   return 0;
 }
 
+/* Ensure the event loop exits with a clear error if acquiring the well-known service name fails */
+static int request_name_callback(sd_bus_message *m, void *userdata, sd_bus_error *ret_error) {
+  if (sd_bus_message_is_method_error(m, NULL)) {
+    int r = sd_bus_message_get_errno(m);
+
+    if (r == ETIMEDOUT)
```

Comment:
hmm, is this a real thing?

Reply:
so i figure it is a real thing. But i figure the right fix is to move it to a match handler for the Connected signal, so that the timeout is not triggered by the "offline" phase

Reply:
I don't understand what you mean? The goal here is to catch cases where the dbus policy was not installed and things like that - it can never work, so it just endlessly loops. The timeout case is the one we want to retry, as that happens when the broker is not up yet. How does the Connected signal help for this?

Reply:
oh, instead of comparing sd_bus_message_get_errno() with ETIMEDOUT please use the bus error natively. Or in other words, the 2nd argument of sd_bus_message_is_method_error() set to SD_BUS_ERROR_TIMEOUT rather than NULL as you do right now.

The translation of the dbus errors to errnos is lossy and not reversible. It also maps multiple error strings to the same error number. Hence you should only use it if you really need an errno. Otherwise just work directly with dbus errors, its more expressive. it's also shorter.

Reply:
> oh, instead of comparing sd_bus_message_get_errno() with ETIMEDOUT please use the bus error natively. Or in other words, the 2nd argument of sd_bus_message_is_method_error() set to SD_BUS_ERROR_TIMEOUT rather than NULL as you do right now.
> 
> The translation of the dbus errors to errnos is lossy and not reversible. It also maps multiple error strings to the same error number. Hence you should only use it if you really need an errno. Otherwise just work directly with dbus errors, its more expressive. it's also shorter.

ok

Reply:
but if the connection goes away, then you get Disconnected signal, destroy the connection, create a new one, and then acquire the name again in the Connected signal and so on, until all eternity.

Reply:
Which is the same, but with an extra step, so I don't see the point